### PR TITLE
Docs - date_sub

### PR DESCRIPTION
**Pull request by bharatnc**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation (changelog entry is not required)

Detailed description / Documentation draft:

Document `date_sub`. Was missed out in:  #20107.

Relates to: #16049

### CODE REVIEW

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 1]
```diff
@@ -380,7 +380,7 @@ Alias: `dateTrunc`.
 
 **Parameters**
 
--   `unit` — The type of interval to truncate the result. [String](../../sql-reference/data-types/string.md).
+-   `unit` — The type of interval to truncate the result. [String Literal](../syntax.md#syntax-string-literal).
```
Comment (bharatnc) [inline docs/en/sql-reference/functions/date-time-functions.md:383]: Minor unrelated change which I overlooked this one in my last PR. This is actually string literal as the `unit` is specified as `'hour'` , `'minute'`, `'second'` etc.

Example:
```sql
ELECT now(), date_trunc('hour', now(), 'Europe/Moscow')
```

**Comment by bharatnc** [comment]
cc: @den-crane @vdimir

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 2]
```diff
@@ -518,6 +518,46 @@ Result:
 └────────────────────────────────────────────────────────────────────────────────────────┘
 ```
 
+## date\_sub {#date_sub}
+
+This subtracts a time/date interval from a date and then returns the date.
```
Comment (vdimir) [inline docs/en/sql-reference/functions/date-time-functions.md:523]:
```suggestion
Subtracts a time/date interval from the provided date.
```

File: docs/en/sql-reference/functions/date-time-functions.md [hunk 3]
```diff
@@ -518,6 +518,46 @@ Result:
...
+
+Aliases: `dateSub`, `DATE_SUB`. 
+
+**Parameters**
+
+-   `unit` — The type of interval to subtract. [String](../../sql-reference/data-types/string.md).
+
+        Supported values: second, minute, hour, day, week, month, quarter, year.
+-   `value` - Value in specified unit - [Int](../../sql-reference/data-types/int-uint.md)    
+-   `date` — [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md).
```
Comment (vdimir) [inline docs/en/sql-reference/functions/date-time-functions.md:539]:
```suggestion
-   `date` — [Date](../../sql-reference/data-types/date.md) or [DateTime](../../sql-reference/data-types/datetime.md) to subtract value from.
```

**Comment by bharatnc** [comment]
Thank you.
